src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -91,6 +91,45 @@ public static string GetFullPath(string path)
             return fullPath;
         }
 
+        public static string GetFullPath(string path, string basePath)
+        {
+            if (basePath == null)
+                throw new ArgumentNullException(nameof(basePath));
+
+            if (!IsPathFullyQualified(basePath))
+                throw new ArgumentException(SR.Arg_BasePathNotFullyQualified);
+
+            if (IsPathFullyQualified(path))
+                return GetFullPath(path);
+
+            int length = path.Length;
+
+            if (PathInternal.IsDevice(basePath))
+            {
+                return RemoveRelativeSegments(CombineNoChecks(basePath, path), PathInternal.GetRootLength(basePath));
+            }
+            else if ((length >= 1 && PathInternal.IsDirectorySeparator(path[0])))
+            {
+                return RemoveRelativeSegments(CombineNoChecks(GetPathRoot(basePath), path.Substring(1)));
+            }
+            else if (length >= 2 && PathInternal.IsValidDriveChar(path[0]) && path[1] == PathInternal.VolumeSeparatorChar)
+            {
+                if (length == 2 || !PathInternal.IsDirectorySeparator(path[2]))
```

Comment:
This line should just be an assert, it should never not be true.

Reply:
what if basepath = "c:\foo" and path = "d:\bar\tmp\..\"  Wont then we will require this pathway ?

Reply:
It would be fully qualified and shouldn't have gotten past the earlier check.  `@"d:bar\tmp"` would get through, and that is our special case (drive relative).